hw/ip/otbn/dv/otbnsim/sim/state.py

```diff
@@ -66,9 +73,41 @@ def set_next_pc(self, next_pc: int) -> None:
         assert(self.is_pc_valid(next_pc))
         self._pc_next_override = next_pc
 
-    def set_rnd_data(self, rnd_data: int) -> None:
+    def step_edn(self, rnd_data: int) -> None:
+        # Take the new data
         self._new_rnd_data = rnd_data
 
+        # Collect 32b packages in a 256b variable
+        self.rnd_256b = ((self.rnd_256b << 32) + self._new_rnd_data)
+        self.rnd_256b = self.rnd_256b & ((1 << 256) - 1)
+        self._new_rnd_data = None
+
+        # Count until 8 valid packages are received
+        self.rnd_256b_counter += 1
+        if self.rnd_256b_counter < 8:
+            return
+
+        # Reset the 32b package counter and wait until receiving done
+        # signal from RTL
+        self.rnd_256b_counter = 0
+
+        self.rnd_cdc_pending = True
```

Comment:
I agree that this is correct, but it's a bit odd to read. Maybe it would be easier if the "exceptional" case went first? So something like this, maybe?
```suggestion
        # Count up to 8 packets. When done, reset the counter and wait until
        # we see that CDC has finished.
        if self.rnd_256b_counter == 8:
            self.rnd_cdc_pending = True
            self.rnd_256b_counter = 0
        else:
            self.rnd_256b_counter += 1
```